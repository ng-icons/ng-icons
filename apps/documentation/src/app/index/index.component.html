<div
  appFadeInContainer
  class="flex flex-col items-center justify-center h-screen px-4 mt-6 md:mt-0"
>
  <img
    appFadeIn
    class="w-32 mb-6"
    src="assets/angular.svg"
    alt="Angular Logo"
  />

  <h2
    appFadeIn
    class="heading mb-5 text-base md:text-xl font-extrabold uppercase"
  >
    Angular Icons
  </h2>

  <h1
    appFadeIn
    class="slogan font-extrabold text-3xl md:text-5xl text-center mb-6 md:leading-[60px]"
  >
    The all-in-one icon<br />
    library for Angular
  </h1>

  <p appFadeIn class="text-sm md:text-base text-[#F5EAEA] text-center mb-16">
    Providing over 48,000 icons for use in your applications
  </p>

  <div appFadeIn class="flex gap-6 flex-col md:flex-row">
    <button
      routerLink="/getting-started"
      class="btn btn-white h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Get Started</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>

    <button
      routerLink="/browse-icons"
      class="btn btn-red h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Browse Icons</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>
  </div>
</div>
