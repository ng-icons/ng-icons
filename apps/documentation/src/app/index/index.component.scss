.heading {
  line-height: 25px;
  letter-spacing: 0.08em;
  background: linear-gradient(360deg, #ffafbd 0%, #ffc3a0 82%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.slogan {
  line-height: 60px;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    rgba(227, 153, 153, 0.96) 100%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.btn {
  border-radius: 32px;
  box-shadow: 0 0 0 rgba(0, 0, 0, 0);

  &.btn-white {
    background: linear-gradient(90deg, #e0eafc 0%, #cfdef3 100%);
    border-top: 1px solid #ffffff;
    color: #1a1e2a;

    &:hover {
      box-shadow: 0px 1px 40px rgba(224, 234, 252, 0.4);
    }
  }

  &.btn-red {
    border-top: 1px solid rgba(238, 238, 238, 0.3);
    background: linear-gradient(90deg, #e53935 0%, #e35d5b 100%);
    color: #fff;

    &:hover {
      box-shadow: 0px 1px 40px rgba(216, 24, 47, 0.5);
    }
  }
}
