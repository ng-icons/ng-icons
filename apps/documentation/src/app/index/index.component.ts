import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';
import { NgIconComponent, provideIcons } from '@ng-icons/core';
import { heroChevronRightMini } from '@ng-icons/heroicons/mini';
import { FadeInContainerDirective } from '../directives/fade-in/fade-in-container.directive';
import { FadeInDirective } from '../directives/fade-in/fade-in.directive';

@Component({
  selector: 'app-index',
  templateUrl: './index.component.html',
  styleUrls: ['./index.component.scss'],
  standalone: true,
  imports: [
    NgIconComponent,
    RouterLink,
    FadeInContainerDirective,
    FadeInDirective,
  ],
  providers: [provideIcons({ heroChevronRightMini })],
})
export class IndexComponent {}
