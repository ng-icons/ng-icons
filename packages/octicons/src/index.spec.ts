import * as icons from './index';

describe('Octicons Icons', () => {
  it('should ensure the export names have not changed unexpectedly', () => {
    expect(Object.keys(icons)).toMatchSnapshot();
  })
});
