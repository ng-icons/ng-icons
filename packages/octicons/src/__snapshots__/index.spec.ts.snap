// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Octicons Icons should ensure the export names have not changed unexpectedly 1`] = `
Array [
  "octAccessibility",
  "octAccessibilityInset",
  "octAlert",
  "octApps",
  "octArchive",
  "octArrowBoth",
  "octArrowDown",
  "octArrowLeft",
  "octArrowRight",
  "octArrowSwitch",
  "octArrowUp",
  "octBeaker",
  "octBell",
  "octBellFill",
  "octBellSlash",
  "octBlocked",
  "octBold",
  "octBook",
  "octBookmark",
  "octBookmarkSlash",
  "octBriefcase",
  "octBroadcast",
  "octBrowser",
  "octBug",
  "octCache",
  "octCalendar",
  "octCheck",
  "octCheckCircle",
  "octCheckCircleFill",
  "octCheckbox",
  "octChecklist",
  "octChevronDown",
  "octChevronLeft",
  "octChevronRight",
  "octChevronUp",
  "octCircle",
  "octCircleSlash",
  "octClock",
  "octCloud",
  "octCloudOffline",
  "octCode",
  "octCodeOfConduct",
  "octCodeReview",
  "octCodeSquare",
  "octCodescan",
  "octCodescanCheckmark",
  "octCodespaces",
  "octColumns",
  "octCommandPalette",
  "octComment",
  "octCommentDiscussion",
  "octContainer",
  "octCopilot",
  "octCopilotError",
  "octCopilotWarning",
  "octCopy",
  "octCpu",
  "octCreditCard",
  "octCrossReference",
  "octDash",
  "octDatabase",
  "octDependabot",
  "octDesktopDownload",
  "octDeviceCamera",
  "octDeviceCameraVideo",
  "octDeviceDesktop",
  "octDeviceMobile",
  "octDiamond",
  "octDiff",
  "octDiffAdded",
  "octDiffIgnored",
  "octDiffModified",
  "octDiffRemoved",
  "octDiffRenamed",
  "octDot",
  "octDotFill",
  "octDownload",
  "octDuplicate",
  "octEllipsis",
  "octEye",
  "octEyeClosed",
  "octFeedDiscussion",
  "octFeedForked",
  "octFeedHeart",
  "octFeedMerged",
  "octFeedPerson",
  "octFeedRepo",
  "octFeedRocket",
  "octFeedStar",
  "octFeedTag",
  "octFeedTrophy",
  "octFile",
  "octFileAdded",
  "octFileBadge",
  "octFileBinary",
  "octFileCode",
  "octFileDiff",
  "octFileDirectory",
  "octFileDirectoryFill",
  "octFileDirectoryOpenFill",
  "octFileMoved",
  "octFileRemoved",
  "octFileSubmodule",
  "octFileSymlinkFile",
  "octFileZip",
  "octFilter",
  "octFlame",
  "octFold",
  "octFoldDown",
  "octFoldUp",
  "octGear",
  "octGift",
  "octGitBranch",
  "octGitCommit",
  "octGitCompare",
  "octGitMerge",
  "octGitMergeQueue",
  "octGitPullRequest",
  "octGitPullRequestClosed",
  "octGitPullRequestDraft",
  "octGlobe",
  "octGrabber",
  "octGraph",
  "octHash",
  "octHeading",
  "octHeart",
  "octHeartFill",
  "octHistory",
  "octHome",
  "octHorizontalRule",
  "octHourglass",
  "octHubot",
  "octIdBadge",
  "octImage",
  "octInbox",
  "octInfinity",
  "octInfo",
  "octIssueClosed",
  "octIssueDraft",
  "octIssueOpened",
  "octIssueReopened",
  "octItalic",
  "octIterations",
  "octKebabHorizontal",
  "octKey",
  "octKeyAsterisk",
  "octLaw",
  "octLightBulb",
  "octLink",
  "octLinkExternal",
  "octListOrdered",
  "octListUnordered",
  "octLocation",
  "octLock",
  "octLog",
  "octLogoGist",
  "octLogoGithub",
  "octMail",
  "octMarkGithub",
  "octMarkdown",
  "octMegaphone",
  "octMention",
  "octMeter",
  "octMilestone",
  "octMirror",
  "octMoon",
  "octMortarBoard",
  "octMultiSelect",
  "octMute",
  "octNoEntry",
  "octNorthStar",
  "octNote",
  "octNumber",
  "octOrganization",
  "octPackage",
  "octPackageDependencies",
  "octPackageDependents",
  "octPaintbrush",
  "octPaperAirplane",
  "octPaperclip",
  "octPaste",
  "octPencil",
  "octPeople",
  "octPerson",
  "octPersonAdd",
  "octPersonFill",
  "octPin",
  "octPlay",
  "octPlug",
  "octPlus",
  "octPlusCircle",
  "octProject",
  "octPulse",
  "octQuestion",
  "octQuote",
  "octReply",
  "octRepo",
  "octRepoClone",
  "octRepoDeleted",
  "octRepoForked",
  "octRepoLocked",
  "octRepoPull",
  "octRepoPush",
  "octRepoTemplate",
  "octReport",
  "octRocket",
  "octRows",
  "octRss",
  "octRuby",
  "octScreenFull",
  "octScreenNormal",
  "octSearch",
  "octServer",
  "octShare",
  "octShareAndroid",
  "octShield",
  "octShieldCheck",
  "octShieldLock",
  "octShieldSlash",
  "octShieldX",
  "octSidebarCollapse",
  "octSidebarExpand",
  "octSignIn",
  "octSignOut",
  "octSingleSelect",
  "octSkip",
  "octSliders",
  "octSmiley",
  "octSortAsc",
  "octSortDesc",
  "octSquare",
  "octSquareFill",
  "octSquirrel",
  "octStack",
  "octStar",
  "octStarFill",
  "octStop",
  "octStopwatch",
  "octStrikethrough",
  "octSun",
  "octSync",
  "octTabExternal",
  "octTable",
  "octTag",
  "octTasklist",
  "octTelescope",
  "octTelescopeFill",
  "octTerminal",
  "octThreeBars",
  "octThumbsdown",
  "octThumbsup",
  "octTools",
  "octTrash",
  "octTriangleDown",
  "octTriangleLeft",
  "octTriangleRight",
  "octTriangleUp",
  "octTrophy",
  "octTypography",
  "octUnfold",
  "octUnlock",
  "octUnmute",
  "octUnverified",
  "octUpload",
  "octVerified",
  "octVersions",
  "octVideo",
  "octWebhook",
  "octWorkflow",
  "octX",
  "octXCircle",
  "octXCircleFill",
  "octZap",
]
`;

exports[`Octicons Icons should ensure the export names have not changed unexpectedly 2`] = `
Array [
  "octAlertLarge",
  "octArchiveLarge",
  "octArrowBothLarge",
  "octArrowDownLarge",
  "octArrowDownLeftLarge",
  "octArrowDownRightLarge",
  "octArrowLeftLarge",
  "octArrowRightLarge",
  "octArrowSwitchLarge",
  "octArrowUpLarge",
  "octArrowUpLeftLarge",
  "octArrowUpRightLarge",
  "octBeakerLarge",
  "octBellLarge",
  "octBellFillLarge",
  "octBellSlashLarge",
  "octBlockedLarge",
  "octBoldLarge",
  "octBookLarge",
  "octBookmarkLarge",
  "octBookmarkFillLarge",
  "octBookmarkSlashLarge",
  "octBookmarkSlashFillLarge",
  "octBriefcaseLarge",
  "octBroadcastLarge",
  "octBrowserLarge",
  "octBugLarge",
  "octCalendarLarge",
  "octCheckLarge",
  "octCheckCircleLarge",
  "octCheckCircleFillLarge",
  "octCheckboxLarge",
  "octChecklistLarge",
  "octChevronDownLarge",
  "octChevronLeftLarge",
  "octChevronRightLarge",
  "octChevronUpLarge",
  "octCircleLarge",
  "octCircleSlashLarge",
  "octClockLarge",
  "octCloudLarge",
  "octCloudOfflineLarge",
  "octCodeLarge",
  "octCodeOfConductLarge",
  "octCodeReviewLarge",
  "octCodeSquareLarge",
  "octCodescanLarge",
  "octCodescanCheckmarkLarge",
  "octCodespacesLarge",
  "octColumnsLarge",
  "octCommandPaletteLarge",
  "octCommentLarge",
  "octCommentDiscussionLarge",
  "octCommitLarge",
  "octContainerLarge",
  "octCopilotLarge",
  "octCopyLarge",
  "octCpuLarge",
  "octCreditCardLarge",
  "octCrossReferenceLarge",
  "octDashLarge",
  "octDatabaseLarge",
  "octDependabotLarge",
  "octDesktopDownloadLarge",
  "octDeviceCameraVideoLarge",
  "octDeviceDesktopLarge",
  "octDeviceMobileLarge",
  "octDiamondLarge",
  "octDiffLarge",
  "octDotLarge",
  "octDotFillLarge",
  "octDownloadLarge",
  "octDuplicateLarge",
  "octEyeLarge",
  "octEyeClosedLarge",
  "octFileLarge",
  "octFileBinaryLarge",
  "octFileCodeLarge",
  "octFileDiffLarge",
  "octFileDirectoryLarge",
  "octFileDirectoryFillLarge",
  "octFileMediaLarge",
  "octFileSubmoduleLarge",
  "octFileSymlinkFileLarge",
  "octFileZipLarge",
  "octFilterLarge",
  "octFlameLarge",
  "octFoldLarge",
  "octFoldDownLarge",
  "octFoldUpLarge",
  "octGearLarge",
  "octGiftLarge",
  "octGitBranchLarge",
  "octGitCommitLarge",
  "octGitCompareLarge",
  "octGitMergeLarge",
  "octGitPullRequestLarge",
  "octGitPullRequestClosedLarge",
  "octGitPullRequestDraftLarge",
  "octGlobeLarge",
  "octGrabberLarge",
  "octGraphLarge",
  "octHashLarge",
  "octHeadingLarge",
  "octHeartLarge",
  "octHeartFillLarge",
  "octHistoryLarge",
  "octHomeLarge",
  "octHomeFillLarge",
  "octHorizontalRuleLarge",
  "octHourglassLarge",
  "octHubotLarge",
  "octImageLarge",
  "octInboxLarge",
  "octInfinityLarge",
  "octInfoLarge",
  "octIssueClosedLarge",
  "octIssueDraftLarge",
  "octIssueOpenedLarge",
  "octIssueReopenedLarge",
  "octItalicLarge",
  "octIterationsLarge",
  "octKebabHorizontalLarge",
  "octKeyLarge",
  "octLawLarge",
  "octLightBulbLarge",
  "octLinkLarge",
  "octLinkExternalLarge",
  "octListOrderedLarge",
  "octListUnorderedLarge",
  "octLocationLarge",
  "octLockLarge",
  "octMailLarge",
  "octMegaphoneLarge",
  "octMentionLarge",
  "octMilestoneLarge",
  "octMirrorLarge",
  "octMoonLarge",
  "octMortarBoardLarge",
  "octMultiSelectLarge",
  "octMuteLarge",
  "octNoEntryLarge",
  "octNorthStarLarge",
  "octNoteLarge",
  "octNumberLarge",
  "octOrganizationLarge",
  "octPackageLarge",
  "octPackageDependenciesLarge",
  "octPackageDependentsLarge",
  "octPaperAirplaneLarge",
  "octPaperclipLarge",
  "octPasteLarge",
  "octPencilLarge",
  "octPeopleLarge",
  "octPersonLarge",
  "octPersonAddLarge",
  "octPersonFillLarge",
  "octPinLarge",
  "octPlayLarge",
  "octPlugLarge",
  "octPlusLarge",
  "octPlusCircleLarge",
  "octProjectLarge",
  "octPulseLarge",
  "octQuestionLarge",
  "octQuoteLarge",
  "octReplyLarge",
  "octRepoLarge",
  "octRepoForkedLarge",
  "octRepoLockedLarge",
  "octRepoPushLarge",
  "octRepoTemplateLarge",
  "octReportLarge",
  "octRocketLarge",
  "octRowsLarge",
  "octRssLarge",
  "octRubyLarge",
  "octScreenFullLarge",
  "octScreenNormalLarge",
  "octSearchLarge",
  "octServerLarge",
  "octShareLarge",
  "octShareAndroidLarge",
  "octShieldLarge",
  "octShieldCheckLarge",
  "octShieldLockLarge",
  "octShieldXLarge",
  "octSidebarCollapseLarge",
  "octSidebarExpandLarge",
  "octSignInLarge",
  "octSignOutLarge",
  "octSingleSelectLarge",
  "octSkipLarge",
  "octSmileyLarge",
  "octSortAscLarge",
  "octSortDescLarge",
  "octSquareLarge",
  "octSquareFillLarge",
  "octSquirrelLarge",
  "octStackLarge",
  "octStarLarge",
  "octStarFillLarge",
  "octStopLarge",
  "octStopwatchLarge",
  "octStrikethroughLarge",
  "octSunLarge",
  "octSyncLarge",
  "octTabLarge",
  "octTableLarge",
  "octTagLarge",
  "octTasklistLarge",
  "octTelescopeLarge",
  "octTelescopeFillLarge",
  "octTerminalLarge",
  "octThumbsdownLarge",
  "octThumbsupLarge",
  "octToolsLarge",
  "octTrashLarge",
  "octTriangleDownLarge",
  "octTriangleLeftLarge",
  "octTriangleRightLarge",
  "octTriangleUpLarge",
  "octTrophyLarge",
  "octTypographyLarge",
  "octUnfoldLarge",
  "octUnlockLarge",
  "octUnmuteLarge",
  "octUnverifiedLarge",
  "octUploadLarge",
  "octVerifiedLarge",
  "octVersionsLarge",
  "octVideoLarge",
  "octWorkflowLarge",
  "octXLarge",
  "octXCircleLarge",
  "octXCircleFillLarge",
  "octZapLarge",
]
`;
